### PR TITLE
Add use-cache parameter to leverage Github cache for v2

### PR DESCRIPTION
**Pull request by davidbuzinski**
* Add a parameter `use-cache` that defaults to false to use GitHub's cache
* save cache as part of a post workflow job on successful runs when parameter is true
* restore cache and no-op the install if parameter is true and there is a cache hit

### CODE REVIEW

File: action.yml
Comment (sameagen-MW) [inline action.yml:21]: Is this going to visible to users if we were to create a release?

File: package.json
Comment (sameagen-MW) [inline package.json:12]: Just to make sure I understand, the contents of `post.ts` are what's getting built into `dist/cache-save/index.js`?

File: action.yml
Comment (davidbuzinski) [reply action.yml]: Oops, I forgot to fill that in with a real description. Fixed

File: package.json
Comment (davidbuzinski) [reply package.json:12]: Yes, `-o dist/cache-save ` specifies the output directory and the default name is always `index.js`. We need this step to convert `post.ts` from typescript to javascript and "minify" everything.

We didn't have to specify as many options before because we were leveraging defaults but now we need to make 2 javascript files: one for the main action and one for the post step. I updated some other stuff in our package.json to be more similar to what the Github supported actions  are doing in these cases.

File: src/properties.json
Comment (davidbuzinski) [inline src/properties.json:4]: Will change back before merging. 2 integ tests fail because of the current mpm issues with Windows & Mac.

File: action.yml
Comment (mcafaro) [inline action.yml:19]: It seems like this is just `cache` in a number of GitHub's own `setup-*` actions (`setup-dotnet`,`setup-go`,`setup-java` (not a boolean),`setup-node` (not a boolean)). Better to follow their convention?

File: action.yml
Comment (mcafaro) [inline action.yml:21]: @mw-hrastega should take a look at this.

File: src/cache-restore.ts
Comment (mcafaro) [inline src/cache-restore.ts:10]: Should the products list be sorted before creating this hash?

File: src/index.ts
Comment (mcafaro) [inline src/index.ts:14]: [`core.getBooleanInput`](https://github.com/actions/toolkit/blob/0d638344746a652dedde8baa2ac04a10e0e0a53d/packages/core/src/core.ts#L173)?

File: src/cache-save.ts
Comment (mcafaro) [inline src/cache-save.ts:10]: This feels a little funny. It seems like `useCache` wants to be a boolean rather than a string. And this function shouldn't even be called if it is false.

File: src/post.ts
Comment (mcafaro) [inline src/post.ts:8]: Get boolean input and don't call `cacheMATLAB` if it is not true?

File: src/index.ts
Comment (davidbuzinski) [reply src/index.ts]: missed `getBooleanInput` somehow

File: action.yml
Comment (davidbuzinski) [reply action.yml]: Changed to `cache` for now

File: src/cache-restore.ts
Comment (davidbuzinski) [reply src/cache-restore.ts]: Updated

File: src/cache-restore.ts
Comment (mcafaro) [inline src/cache-restore.ts:16]: Why is this an array rather than a scalar?

File: src/cache-save.ts
Comment (mcafaro) [inline src/cache-save.ts:12]: When would this be empty?

File: src/cache-restore.ts
Comment (davidbuzinski) [reply src/cache-restore.ts]: Changed to scalar because it makes more sense. Originally wanted to check / convert once in cache-save before passing it to cache.saveCache (takes string array). I might've been trying to be "too safe" in the `cache-save` code when getting the value from state.

File: src/cache-save.ts
Comment (davidbuzinski) [reply src/cache-save.ts]: It realistically shouldn't be. Took out the "unwrapping" stuff and changed it to a scalar.

**Review by mcafaro — APPROVED** [review]

File: action.yml
Comment (mw-hrastega) [reply action.yml]: Option to use a cache for optimizing the product installation time, specified as false or true?

(Note: A doc style revision now requires us to use "Option to" instead of "Whether to". If the file you are revising has other instances of "Whether to", please consider updating them for consistency.)

Comment (mw-hrastega) [reply action.yml]: a cache --> the GitHub Actions cache

Comment (davidbuzinski) [reply action.yml]: "Option to store the MATLAB installation in the GitHub Actions cache, specified as false or true."

How does this sound? I am questioning my use of the word optimize...

Comment (mw-hrastega) [reply action.yml]: I am in favor of removing optimize.

1) Can "the MATLAB installation" be simplified as "MATLAB"? Or are these two different things?
2) Is this only for MATLAB and not other MathWorks products?

Comment (davidbuzinski) [reply action.yml]: "Option to store MATLAB in the GitHub Actions cache, specified as false or true."

I think this makes sense even in the case of point 2 below

1. Yes sounds good
2. It is technically for all of our products that can be installed with the MATLAB installer (MATLAB, Simulink, etc.) and possibly support packages at some point.

**Review by mw-hrastega — APPROVED** [review]

**Pull request merged**
